contractsV5/FeeHandler.sol

```diff
@@ -0,0 +1,200 @@
+        uint amount = totalRewardsPerEpoch[epoch] * percentageInPrecision / PRECISION;
+
+        // Update total rewards and total rewards per epoch
+        totalRewardsPerEpoch[epoch] -= amount;
+        totalRewards -= amount;
+
+        // send reward to staker
+        (bool success, ) = staker.call.value(amount)("");
+        require(success, "Transfer of rewards to staker failed.");
+
+        emit DistributeRewards(staker, amount);
+
+        return amount;
+    }
+
+    event DistributeRebate(address rebateWallet, uint amountWei);
+
+    // Using rebateWallet instead of reserve so I don't have to store KyberNetwork variable.
+    function claimReserveRebate(address rebateWallet) public returns (uint){
+        // Get total amount of rebate accumulated
+        uint amount = totalRebatesPerRebateWallet[rebateWallet] - 1;
+
+        // Update total rebate and rebate per rebate wallet amounts
+        totalRebates -= amount;
+        totalRebatesPerRebateWallet[rebateWallet] = 1; // Do not use 0 to avoid paying high gas when setting back from 0 to non zero.
+
+        // send rebate to rebate wallet
+        (bool success, ) = rebateWallet.call.value(amount)("");
+        require(success, "Transfer of rebates to rebate wallet failed.");
+
+        emit DistributeRebate(rebateWallet, amount);
+
+        return amount;
+    }
+
+    // we will have to limit amounts. per burn.
+    // and create some block delay between burns.
+    // Todo: include arbitrage check https://github.com/KyberNetwork/smart-contracts/pull/433/files
+    function burnKNC() public returns(uint) {
+        // check if current block > last buy block number + num block interval
+        require(
+            block.number < lastBurnBlock + burnBlockInterval,
+            "Unable to buy as burnBlockInterval has not passed since lastBurnBlock"
+        );
+
+        // update last buy block number
+        lastBurnBlock = block.number;
+
+        // Get srcQty to burn, if greater than ETH_TO_BURN, burn only ETH_TO_BURN per function call.
+        uint srcQty = address(this).balance - totalRebates - totalRewards;
+        srcQty = srcQty > ETH_TO_BURN ? ETH_TO_BURN : srcQty;
+
+        // Get the slippage rate
+        // If srcQty is too big, get expected rate will return 0 so maybe we should limit how much can be bought at one time.
+        uint expectedRate = kyberNetworkProxyContract.getExpectedRateAfterCustomFee(IERC20(ETH_TOKEN_ADDRESS), IERC20(knc), srcQty, 0, "");
+
+        // Buy some KNC and burn
+        uint destQty = kyberNetworkProxyContract.tradeWithHintAndPlatformFee(
+            ETH_TOKEN_ADDRESS,
+            srcQty,
+            IERC20(knc),
+            address(this),
+            MAX_QTY,
+            expectedRate * 97 / 100,
+            address(0), // platform wallet
+            0, // platformFeeBps
+            "" // hint
+        );
+
+        // Burn KNC
+        require(IBurnableToken(knc).burn(destQty), "KNC burn failed");
```

Comment:
can define knc as burnable token in the first place. so don't have to cast it.

Reply:
noted and updated